pkg/provisioner/controller.go

```diff
@@ -380,7 +470,43 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 		return fmt.Errorf("error getting reference to OBC: %v", err)
 	}
 
-	// create Secret and ConfigMap
+	// Create auth Secret and bucket info ConfigMap
+	// If the operator crashes before the OB is created, provisioning will happen again including a
+	// call to Provision()/Grant(), and a Secret and/or ConfigMap might exist from the previous
+	// provisioning attempt. Provision() isn't guaranteed to give the same credentials or bucket
+	// each time, so if the Secret/ConfigMap already exists, delete the old one and re-create it
+	// with the latest info.
+
+	secret, err = getExistingSecret(obc, c.clientset, defaultRetryBaseInterval, defaultRetryTimeout)
```

Comment:
Can all of this secret and CM handling be contained in helper funcs, just to improve readability by shortening this func?